README.md

```diff
@@ -137,82 +137,179 @@ The trace configuration additionally exposes the `samplingRate` option which set
 
-For any of the web frameworks listed above (`express`, `hapi`, and `restify`), a root span is automatically started whenever an incoming request is received (in other words, all middleware already runs within a root span). If you wish to record a span outside of any of these frameworks, any traced code must run within a root span that you create yourself.
+For any of the web frameworks listed above (`express`, `hapi`, `koa` and `restify`), a root span is automatically started whenever an incoming request is received (in other words, all middleware already runs within a root span). If you wish to record a span outside of any of these frameworks, any traced code must run within a root span that you create yourself.
 
-The API is exposed by the `agent` returned by a call to `start`:
+### Accessing the API
 
-```javascript
-  var agent = require('@google/cloud-trace').start();
-```
-
-For child spans, you can either use the `startSpan` and `endSpan` API, or use the `runInSpan` function that uses a callback-style. For root spans, you must use `runInRootSpan`.
-
-### Start & end
-
-To start a new child span, use `agent.startSpan`. Each span requires a name, and you can optionally specify labels.
+Calling the `start` function returns an instance of `TraceApi`, which provides an interface for tracing:
 
 ```javascript
-  var span = agent.startSpan('name', {label: 'value'});
+  var traceApi = require('@google/cloud-trace').start();
 ```
 
-Once your work is complete, you can end a child span with `agent.endSpan`. You can again optionally associate labels with the span:
+It can also be retrieved by subsequent calls to `get` elsewhere:
 
 ```javascript
-  agent.endSpan(span, {label2: 'value'});
+  // after start() is called
+  var traceApi = require('@google/cloud-trace').get();
 ```
 
-Note that the labels parameter must be an object. Other types (e.g. string) will be silently ignored.
+The object returned by both of these calls is guaranteed to have the surface described below, even if the agent is disabled.
+
+### The `TraceApi` Object
+
+A `TraceApi` instance, in short, provides functions that facilitate the following:
+
+- Creating trace spans and add labels to them.
+- Getting information about how the trace agent was configured in the current application.
+- Parsing and serializing trace contexts to support distributed tracing between microservices.
+- Binding callbacks and event emitters in order to propagate trace contexts across asynchronous boundaries.
+
+In addition to the above, `TraceApi` also provides a number of well-known label keys and constants through its `labels` and `constants` fields respectively.
+
+#### Trace Spans
+
+These functions provide the capability to create trace spans, add labels to them, and close them. `transaction` and `childSpan` are instances of `Transaction` and `ChildSpan`, respectively.
```

Comment:
Oops, should have checked that references to `transaction` were removed.

I'm going to delete this sentence altogether, since the notation changed and I no longer refer to canonical instances of `*Span` at all.